### DISCUSSION TITLE
Accidentally deleted a bunch of photos after moving a file with auto-rename turned on

### DISCUSSION THREAD

**Discussion by cyberlaner** [trganda/obsidian-attachment-management#59] [Q&A] [answered]
I moved a file which had numerous attachment paths in it, the auto rename function activated, then all of the old photos in that file were empty in the new file.

**Comment by trganda** [comment]
Could you please provide more detail of you problem? I have trid to moved a markdown file with multi-linked images in version 0.8.3, the linked images was not empty.

**Reply by cyberlaner** [reply]
Below are some notes I took trying to recreate the situation.  I could not get the same results to appear in my test repo for this.  

The main aspects of the issue are that:
1. I'm using GitHub with a gitignore file to push changes.  This could've somehow created issues with it, but I didn't ignore

[UPDATE] I think I may have found the issue.  It seems I may have inadvertently ignored the folder.
![image](https://github.com/trganda/obsidian-attachment-management/assets/143642425/4ee6d770-a7eb-4ea7-9165-0df2f8eb6bd7)

[Fixing Attachments Obsidian - Troubleshooting.md](https://github.com/trganda/obsidian-attachment-management/files/12686858/Fixing.Attachments.Obsidian.-.Troubleshooting.md)

**Reply by cyberlaner** [reply]
Then again, these files should've still stayed in my vault, but maybe I'm misunderstanding how this would have interacted with everything.  I'm trying to keep a local wiki synced with one in a repo and your tool to keep images organized into folders that get automatically put into the repo via the gitignore file.

**Reply by cyberlaner** [reply]
Nevermind.  The gitignore doesn't seem like it should ignore those files anyway.  Even so, I'm not sure that moving files into an ignored folder would cause them to be deleted.  I'm doing some more tests with copy and pasting.  I noticed that moving a note, then hitting Ctrl Z did some weird things

**Reply by cyberlaner** [reply]
I even went back to commits where the images existed and shown on my repo history, but then they show up empty in Obsidian...weird. 

![image](https://github.com/trganda/obsidian-attachment-management/assets/143642425/944dab1b-0464-4eef-8dbe-60af73c56fd9)

**Reply by cyberlaner** [reply]
Here's the two files showing up in History in GitHub Desktop

![image](https://github.com/trganda/obsidian-attachment-management/assets/143642425/27b79ba8-1756-4a92-b4f3-82f616401200)

**Reply by cyberlaner** [reply]
Just to confirm, I went a commit back and used "checkout" to look at the files.

**Reply by cyberlaner** [reply]
Okay I found the issue.  it's the gitignore file.  maybe I'll make an obsidian plugin to properly integrate with attachments and show what's being ignored.  *big sad*

**Reply by cyberlaner** [reply]
I'm still confused about why the images aren't showing up though.  I've shown what i mean in the below notes
[Fixing Attachments Obsidian - Troubleshooting.md](https://github.com/trganda/obsidian-attachment-management/files/12691067/Fixing.Attachments.Obsidian.-.Troubleshooting.md)

**Reply by cyberlaner** [reply]
This might be an issue with attachment paths but I'm not sure.  If you have any recommendations for obsidian plugins that make attachment paths more explicit then I'm all ears, but it seems obsidian isn't getting the image, despite "Show in system explorer" showing the actual image.

**Reply by cyberlaner** [reply]
One last screenshot of what the image is linking to 
![image](https://github.com/trganda/obsidian-attachment-management/assets/143642425/e2d14974-880b-4f47-9ea1-6f5f0af8428b)

**Comment by trganda** [comment]
> Below are some notes I took trying to recreate the situation. I could not get the same results to appear in my test repo for this.以下是我尝试重现这种情况时所做的一些笔记。我无法在我的测试仓库中获得相同的结果。
> 
> The main aspects of the issue are that:问题的主要方面是：
> 
> 1. I'm using GitHub with a gitignore file to push changes.  This could've somehow created issues with it, but I didn't ignore我使用 GitHub 和 gitignore 文件来推送更改。这可能会以某种方式产生问题，但我没有忽视
> 
> [UPDATE] I think I may have found the issue. It seems I may have inadvertently ignored the folder.[更新]我想我可能已经发现了这个问题。看来我可能无意中忽略了该文件夹。 ![image](https://user-images.githubusercontent.com/143642425/269684964-4ee6d770-a7eb-4ea7-9165-0df2f8eb6bd7.png)
> 
> [Fixing Attachments Obsidian - Troubleshooting.md 修复附件黑曜石 - Troubleshooting.md](https://github.com/trganda/obsidian-attachment-management/files/12686858/Fixing.Attachments.Obsidian.-.Troubleshooting.md)

I have some question nedd to ask.

The `.gitignore` was added (or updated `!__attachments`, `__attachemnts/**/` ... etc) to this vault after you already used this vault a long time? And before this operation, you already commit some image file under `__attachments`.

The **Rescources & Docs** in this picture was the subfolder of **__attachments** ? 

If all this questoin was yes, the previers commited images will not be ignore, event you have updated `.gitignore` file. And the new added images file under `... & S3 & Logs`  was ignored by `.gitignore` with`__attachments/**/` (ignore all subfolder of __attachments) not `__attachments/*Logs*`

**Reply by cyberlaner** [reply]
I can confirm that the __attachments folder had all the same path names as where the images were.  The __attachments/**/ was actually ignoring the images.  However, it's really the issue of how obsidian wasn't finding the image even though it seems that the path and everything was correct.

**Comment by trganda** [comment]
> Nevermind. The gitignore doesn't seem like it should ignore those files anyway. Even so, I'm not sure that moving files into an ignored folder would cause them to be deleted. I'm doing some more tests with copy and pasting. I noticed that moving a note, then hitting Ctrl Z did some weird things没关系。无论如何， gitignore 似乎不应该忽略这些文件。即便如此，我不确定将文件移动到忽略的文件夹中是否会导致它们被删除。我正在通过复制和粘贴进行更多测试。我注意到移动一个音符，然后按 Ctrl Z 会发生一些奇怪的事情

When you copy a images to a markdonw file in obsidian, the images will be named to `Pasted images ... .png` detault, and add a content of `![](Pasted%20iamges%20... .png)`.
And if you enabled this plugins, it will replace `![](Pasted%20iamges%20... .png)` with the attachment name you setted, like `![](IMG-... .png)`, and rename the `Pasted%20iamges%20... .png` on your disk to `IMG-... .png`.

So, if you press `Ctrl-Z`, the editor will Undo the text replace operation, `![](IMG-... .png)` was retreat to `![](Pasted%20iamges%20... .png)` but on your disk, this file was already renamed to `IMG-... .png`, so obsidian can not found it.

**Reply by cyberlaner** [reply]
It seems that the attachment in Obsidian and the disk part have matching names, so I'm not sure what's happening there.

![image](https://github.com/trganda/obsidian-attachment-management/assets/143642425/d8d366be-3c75-4fa2-8bd7-99cf1721c449)

![image](https://github.com/trganda/obsidian-attachment-management/assets/143642425/42db8653-633a-40e2-a7f6-5be68a1eb02c)

![image](https://github.com/trganda/obsidian-attachment-management/assets/143642425/55eb2c8c-eb78-4ac8-9055-5213128120ab)

**Comment by trganda** [comment]
And the most important thing, this plugin will never delete your attachment, it just rename or move it. So if the attachment was exists but you can not see that in notes file, this should be happen if the attachment was renamed but this plugin (or obsidian, I'm using obsidian api to update link) has no updated the link of the notes.

**Comment by cyberlaner** [comment, marked as answer]
I found the issue here once again.  The issue is the max file path length in Windows. I'm not sure how to get around this without changing windows settings.  Apparently UNC Path or the '\\?\' prefix can help, but I'm not sure of a way to have this and relative file paths at the same time.  

- [Maximum Path Length Limitation - Win32 apps | Microsoft Learn](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry)
